### PR TITLE
ApexActionStatus class Code Coverage 100%

### PR DESCRIPTION
**Pull request by jsNickOs**

### CODE REVIEW

**Review by mykhaylo-partyka — CHANGES_REQUESTED** [review]

File: ss_admin_console/main/tests/classes/testdata/TestInterfaceClass.cls [hunk 1]
```diff
@@ -0,0 +1,3 @@
+public interface TestInterfaceClass {
+    void calculate(double x);
```
Comment (mykhaylo-partyka) [inline ss_admin_console/main/tests/classes/testdata/TestInterfaceClass.cls:2]: what is the method for?

File: ss_admin_console/main/tests/classes/testdata/TestInterfaceClass.cls [hunk 2]
```diff
@@ -0,0 +1,3 @@
+public interface TestInterfaceClass {
```
Comment (mykhaylo-partyka) [inline ss_admin_console/main/tests/classes/testdata/TestInterfaceClass.cls:1]: TestInterfaceClass - is it interface or class?

File: ss_admin_console/main/tests/classes/testdata/TestSubClass1.cls [hunk 3]
```diff
@@ -0,0 +1,5 @@
+public with sharing class TestSubClass1 implements TestInterfaceClass{
```
Comment (mykhaylo-partyka) [inline ss_admin_console/main/tests/classes/testdata/TestSubClass1.cls:1]: Fix formatting

**Review by mykhaylo-partyka — CHANGES_REQUESTED** [review]

File: ss_admin_console/main/tests/classes/SSACUtilTest.cls [hunk 4]
```diff
@@ -29,4 +29,19 @@ public inherited sharing class SSACUtilTest {
         Test.stopTest();
         System.assertNotEquals(null, interfaces, 'JobClasses successfully recovered.');
     }
+
+    @isTest
+    static void testGetSubClasses() {
+        SSACUtil utility = new SSACUtil();
+
+        TestSubClass1 obj1 = new TestSubClass1();
```
Comment (mykhaylo-partyka) [inline ss_admin_console/main/tests/classes/SSACUtilTest.cls:37]: Why do you create instances of TestSubClassX? Where is it used?

File: ss_admin_console/main/tests/classes/testdata/TestInterface.cls [hunk 5]
```diff
@@ -0,0 +1,3 @@
+public interface TestInterface {
+    void calculate(double x);
```
Comment (mykhaylo-partyka) [inline ss_admin_console/main/tests/classes/testdata/TestInterface.cls:2]: What is this method for?

File: ss_admin_console/main/tests/classes/SSACUtilTest.cls [hunk 6]
```diff
@@ -29,4 +29,19 @@ public inherited sharing class SSACUtilTest {
...
+        TestSubClass2 obj2 = new TestSubClass2();
+        TestSubClass3 obj3 = new TestSubClass3();
+
+        Test.startTest();
+        List<Type> subClasses = utility.getSubClasses(TestInterface.class);
+        Test.stopTest();
+
+        System.assertEquals(4, subClasses.size());
```
Comment (mykhaylo-partyka) [inline ss_admin_console/main/tests/classes/SSACUtilTest.cls:45]: Why 4?

**Review by mykhaylo-partyka — CHANGES_REQUESTED** [review]

File: ss_admin_console/main/tests/classes/testdata/TestSubClass3.cls [hunk 7]
```diff
@@ -0,0 +1,5 @@
+public with sharing class TestSubClass3 implements TestInterface {
+    public void calculate (double x) {
+        System.debug('cube of ='+x*x*x);
```
Comment (mykhaylo-partyka) [inline ss_admin_console/main/tests/classes/testdata/TestSubClass3.cls:3]: Formatting!!!

**Review by mykhaylo-partyka — CHANGES_REQUESTED** [review]

File: ss_admin_console/main/tests/classes/testdata/TestInterface.cls [hunk 8]
```diff
@@ -0,0 +1,14 @@
+public interface TestInterface {
+
+    void someMethod(String msg);
+/*
+    public with sharing class TestSubInterfaceClass1 implements TestInterface {
```
Comment (Taraass) [inline ss_admin_console/main/tests/classes/testdata/TestInterface.cls:5]: what are these comments about?

File: ss_admin_console/main/tests/classes/testdata/TestInterfaceSub1.cls [hunk 9]
```diff
@@ -0,0 +1,5 @@
+public with sharing class TestInterfaceSub1 implements TestInterface {
+    public void someMethod(String msg) {
```
Comment (Taraass) [inline ss_admin_console/main/tests/classes/testdata/TestInterfaceSub1.cls:2]:
```suggestion

    public void someMethod(String msg) {
```

File: ss_admin_console/main/tests/classes/testdata/TestInterfaceSub2.cls [hunk 10]
```diff
@@ -0,0 +1,5 @@
+public with sharing class TestInterfaceSub2 implements TestInterface {
+    public void someMethod(String msg) {
```
Comment (Taraass) [inline ss_admin_console/main/tests/classes/testdata/TestInterfaceSub2.cls:2]:
```suggestion

    public void someMethod(String msg) {
```

**Review by Taraass — CHANGES_REQUESTED** [review]

File: ss_admin_console/main/default/classes/utils/SSACUtil.cls [hunk 11]
```diff
@@ -16,9 +16,14 @@ public inherited sharing class SSACUtil {
...
-            if (classType != null && theType.isAssignableFrom(classType)) {
-                result.add(classType);
+            try {
+                Type classType = Type.forName(aClass.Name);
+                if (classType != null && theType.isAssignableFrom(classType) && theType != classType) {
+                    result.add(classType);
+                }
+            }
+            catch (Exception e) {
+                throw e;
```
Comment (mykhaylo-partyka) [inline ss_admin_console/main/default/classes/utils/SSACUtil.cls:26]: What is the idea of rethrowing the exception?

File: ss_admin_console/main/tests/classes/testdata/TestInterface.cls [hunk 12]
```diff
@@ -0,0 +1,3 @@
+public interface TestInterface {
+    void someMethod(String msg);
```
Comment (mykhaylo-partyka) [inline ss_admin_console/main/tests/classes/testdata/TestInterface.cls:2]: Do you need the method?

File: ss_admin_console/main/tests/classes/testdata/TestInterfaceSub1.cls [hunk 13]
```diff
@@ -0,0 +1,4 @@
+public with sharing class TestInterfaceSub1 implements TestInterface {
+
```
Comment (mykhaylo-partyka) [inline ss_admin_console/main/tests/classes/testdata/TestInterfaceSub1.cls:2]: Remove empty line

**Review by mykhaylo-partyka — CHANGES_REQUESTED** [review]

File: ss_admin_console/main/default/classes/utils/SSACUtil.cls (hunk 11 above)
Comment (jsNickOs) [reply ss_admin_console/main/default/classes/utils/SSACUtil.cls:26]: "throw e;" replaced with "System.debug('Error in SSACUtil.getSubClasses: ' + e.getMessage());"

File: ss_admin_console/main/tests/classes/testdata/TestInterface.cls (hunk 12 above)
Comment (jsNickOs) [reply ss_admin_console/main/tests/classes/testdata/TestInterface.cls:2]: I will remove this method

File: ss_admin_console/main/tests/classes/testdata/TestInterfaceSub1.cls (hunk 13 above)
Comment (jsNickOs) [reply ss_admin_console/main/tests/classes/testdata/TestInterfaceSub1.cls:2]: I will remove this empty line

**Review by mykhaylo-partyka — APPROVED** [review]

**Review by Taraass — APPROVED** [review]

**Pull request merged**